File: data/readable_format/fortran/file1452.f90
! sla_DCS2C.f90
SUBROUTINE sla_DCS2C (A, B, V)
  IMPLICIT NONE
  DOUBLE PRECISION A, B, V(3)
  DOUBLE PRECISION COSB
  COSB = COS(B)
  V(1) = COS(A) * COSB
  V(2) = SIN(A) * COSB
  V(3) = SIN(B)
END SUBROUTINE sla_DCS2C

PROGRAM test_sla_DCS2C
  IMPLICIT NONE
  DOUBLE PRECISION V(3)
  DOUBLE PRECISION A, B
  INTEGER :: i

  ! Test case 1
  A = 0.0D0
  B = 0.0D0
  CALL sla_DCS2C(A, B, V)
  PRINT *, "Test case 1: A =", A, "B =", B
  DO i = 1, 3
     PRINT *, "V(", i, ") = ", V(i)
  END DO

  ! Test case 2
  A = 3.14159265358979323846D0 / 4.0D0
  B = 3.14159265358979323846D0 / 4.0D0
  CALL sla_DCS2C(A, B, V)
  PRINT *, "Test case 2: A =", A, "B =", B
  DO i = 1, 3
     PRINT *, "V(", i, ") = ", V(i)
  END DO
END PROGRAM test_sla_DCS2C

File: data/readable_format/cpp/file1452.cpp
// sla_DCS2C.cpp
#include <cmath>
#include <iostream>
#include <cassert>

void sla_DCS2C(double A, double B, double V[3]) {
    double COSB = std::cos(B);
    V[0] = std::cos(A) * COSB;
    V[1] = std::sin(A) * COSB;
    V[2] = std::sin(B);
}

int main() {
    double V[3];

    // Test case 1
    sla_DCS2C(0.0, 0.0, V);
    std::cout << "Test case 1: A = 0 B = 0" << std::endl;
    assert(std::abs(V[0] - 1.0) < 1e-6);
    assert(std::abs(V[1] - 0.0) < 1e-6);
    assert(std::abs(V[2] - 0.0) < 1e-6);

    // Test case 2
    sla_DCS2C(3.14159265358979323846 / 4.0, 3.14159265358979323846 / 4.0, V);
    std::cout << "Test case 2: A = " << 3.14159265358979323846 / 4.0 << " B = " << 3.14159265358979323846 / 4.0 << std::endl;
    assert(std::abs(V[0] - 0.5) < 1e-6);
    assert(std::abs(V[1] - 0.5) < 1e-6);
    assert(std::abs(V[2] - std::sqrt(2)/2) < 1e-6);
    
    std::cout << "All tests passed." << std::endl;
    return 0;
}